vm/vmExpr.go

```diff
@@ -737,12 +737,12 @@ func invokeExpr(expr ast.Expr, env *Env) (reflect.Value, error) {
 	case *ast.DeleteExpr:
 		mapExpr, err := invokeExpr(e.MapExpr, env)
 		if err != nil {
-			return nilValue, newStringError(e, err.Error())
+			return nilValue, newStringError(e.MapExpr, err.Error())
 		}
 
 		keyExpr, err := invokeExpr(e.KeyExpr, env)
 		if err != nil {
-			return nilValue, newStringError(e, err.Error())
+			return nilValue, newStringError(e.KeyExpr, err.Error())
```

Comment:
And this:
return nilValue, newError(e.KeyExpr, err)

Reply:
done.